plenum/server/future_primaries_batch_handler.py

```diff
@@ -0,0 +1,97 @@
+import copy
+from collections import OrderedDict
+from typing import Tuple, Dict
+
+from common.exceptions import LogicError
+from plenum.common.constants import TXN_TYPE, NODE, TARGET_NYM, ALIAS, SERVICES, VALIDATOR, DATA, POOL_LEDGER_ID
+from plenum.common.util import getMaxFailures
+from plenum.server.batch_handlers.batch_request_handler import BatchRequestHandler
+from plenum.server.batch_handlers.three_pc_batch import ThreePcBatch
+
+
+class NodeState:
+    def __init__(self, node_reg, node_ids, primaries):
+        self.node_reg = node_reg
+
+        # Nodes are never delete from here, so they are in a sequential order
+        self.node_ids = node_ids
+
+        # List of future primaries
+        self.primaries = primaries
+
+        self.number_of_inst = FuturePrimariesBatchHandler. \
+            get_required_number_of_instances(len(node_reg))
+
+
+class FuturePrimariesBatchHandler(BatchRequestHandler):
```

Comment:
It would be great to cover the class with more unit tests

Reply:
ok